src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt

```diff
@@ -62,3 +73,68 @@ private class RsLiteralFileReferenceSet(
         }
     }
 }
+
+private val pathValueLiteral: PsiElementPattern.Capture<RsLitExpr> = psiElement<RsLitExpr>()
+    .with("onPathLiteral") { expr -> expr.kind is RsLiteralKind.String && isPathLitExpr(expr) }
+
+private fun isPathLitExpr(expr: RsLitExpr): Boolean {
+    val (function, arguments) = getFunctionAndArguments(expr) ?: return false
+    val owner = function.owner as? RsAbstractableOwner.Impl
+    val ownerType = (owner?.impl?.typeReference?.type as? TyAdt)?.item
+    val (implLookup, knownItems) = expr.implLookupAndKnownItems
+    val isCallExpr = arguments.parent is RsCallExpr
+
+    return when {
+        // Path::new(<literal>)
+        ownerType == knownItems.Path && owner?.isInherent == true && function.name == "new" -> true
```

Comment:
Let's check function name condition first because it can be much more cheaper than `knownItems.Path`